### PR TITLE
Release v1.4.0+51 — Recipes ingredient scanner, fasting timer, water tracker, weight history, Material You, sk locale, and CI/CD pipeline

### LINKED ISSUE #32: Feature Request: Add water tracker
Title says it all.  A way to track how many units of plain water you've drank would fit nicely here.  For a bonus to that feature, you can opt to select drink mixes that you may have added in as well

### PR DESCRIPTION
**Pull request by jordan-lee-code**
> @simonoppowa — this is the next release rolling up everything that's landed on `develop` since `1.3.1+50` (last release PR #381). Once you've had a look, please **merge into `main` and cut a build for the App Store and Play Store** so users get the new water tracker, fasting timer, weight history, Material You theming, recipes ingredient scanner, Quick add, and the rest of the work below. Happy to walk through any section if useful.

## Review feedback addressed (#470)

Thank you for the careful review. Your comments are worked through in #470, which merges into `develop` and so flows up into this release:

- The diary nutrient panel now shows a *limit* qualifier on the ceiling nutrients (sodium, saturated fat, sugar) so they no longer read as targets.
- IOM sources are cited inline: the 2019 sodium/potassium update for the sodium CDRR and potassium AI, and the 2011 calcium/vitamin D report. Those adult calcium and vitamin D figures are the current RDAs, which superseded the 1997 AI values, so they stay as-is with the citation at the line.
- The low-kcal floor comment now cites the Harvard Health reference.
- Notification channel names and descriptions are localized across all nine locales, including the reboot-restore path.
- The "2024 Compendium" attribution is hidden for custom activities.

## Summary

Release **v1.4.0+51** (from `1.3.1+50`). Substantial feature release plus the full CI/CD release pipeline you have been building on `integrate-cicd-deployment` — TestFlight and Play Store internal deployment now run from a green workflow on every release merge. ~83 non-merge commits across ~404 files (+40K / -6.2K LoC overall; +30.2K / -5.8K in `lib/`).

### Water tracker (#418, closes #32)
- Home-screen water chip with quick-add increments and an editable daily goal.
- Profile reflects the same goal and history.
- Undo for the last entry on the home chip.
- New Hive type `WaterIntakeDBO` in its own `WaterIntakeBox`.

### Fasting timer (#432)
- Optional intermittent-fasting timer with a content-warning gate the first time you open it.
- Home chip surfaces the time remaining; tap to start, pause, or reset.
- Completion notification fires when the chosen window elapses (via `flutter_local_notifications`).
- New Hive type `FastingSessionDBO` in its own `FastingBox`.

### Weight history + target weight (#400, #406)
- Capture weight during onboarding and on demand from the profile screen.
- Trend chart on profile with a dashed line at the user's target.
- Optional **calorie taper** as the user approaches the target so the daily kcal budget contracts smoothly rather than jumping when the goal hits.
- New Hive type `WeightLogDBO` in its own `WeightLogBox`.

### Quick add intake (#451)
- "Quick add" text button in the add-meal screen's top app bar.
- Modal bottom sheet with five fields — Title and Energy required, Carbs/Fat/Protein optional.
- Stores a one-off custom `MealEntity` with the entered values mapped onto per-100g nutriments so the existing `IntakeEntity.totalKcal` math returns the exact number the user typed.
- Deliberately not persisted to the custom-meals search index — Quick add entries are numbers, not reusable foods.

### Recipes ingredient picker with barcode scanner (#444)
- Recipe builder gets a dedicated ingredient picker with its own barcode scan flow.
- Adds OFF / FDC products as ingredients directly from the picker rather than bouncing through the main add-meal screen.

### Material You + accent picker (#426)
- Adopts the system accent on Android 12+.
- Picker with **16 built-in presets** for users who prefer to override.
- Persisted per-user.

### Adaptive app icon — iOS dark / tinted + Android themed (#447)
- iOS Asset Catalog covers light, dark, and tinted appearances for every required size (and 26mm visionOS-friendly sizes).
- Android adaptive icon ships a monochrome layer so Android 13+ themed icons pick up the user's wallpaper accent.

### Micronutrient panel with IOM DRI bars (#404, #427, #401)
- Daily and weekly panel for fibre, sodium, saturated fat, sugar, calcium, iron, potassium, vitamin D, vitamin B12, and magnesium.
- Per-nutrient goal can be planned in settings (#401) — falls back to the IOM Dietary Reference Intake when no explicit goal is set.
- Optional DRI bars (#427) draw the reference range on each row so the user can see where they sit against the IOM tables.
- Settings switch hides individual nutrients the user doesn't want to track.

### Per-meal kcal share with presets (#402)
- Settings → Calculations → Meal targets lets the user pick Standard / OMAD / Five-small / Mediterranean / Two-meal, or define a custom share that sums to 100%.
- Each meal section header shows `consumed / target kcal` for that meal once a target is set.

### kcal / kJ energy unit switch (#403)
- Global switch in Settings → Calculations.
- Every diary entry, target, chart, and meal-detail field reflects the choice.
- Edit-meal flow re-interprets the energy field correctly when the user switches mid-edit.

### Daily kcal budget on meal detail (#452)
- Compact banner pinned below the back arrow on the meal detail screen.
- Two-tone primary-green bar: solid up to the kcal already logged today, lighter on top showing what the current selection would add.
- Reflects the day the user is logging against, not just today, so editing a past entry still answers the right question.

### Custom activity type with templates (#408)
- A "Custom" entry in the activity picker lets users log activities that aren't in the Compendium with direct kcal entry.
- Saved entries become reusable templates so common gym sessions don't need re-entering.
- New Hive type `CustomActivityTemplateDBO` in its own `CustomActivityTemplateBox`.

### Custom meals — photo, barcode, Simple form, deduplication
- Attach a photo (gallery or camera) to a custom meal or recipe (#399).
- Attach a barcode to a custom meal so future scans recognise it (#407).
- New **Simple** form mode for direct macro entry without the per-100g maths (#410), with an empty-field guard that saves blanks as 0 rather than null (#430) and no longer multiplies entered totals by 100 (#428).
- One-off meals can be logged without being saved to the custom list (#389).
- Merge duplicate custom foods with successor selection (#429).
- Serving size carries through CSV and JSON import (#424).

### Diary refinements
- Sort meal sections by macro contribution; the preference persists (#405).
- Mirror Home's macro progress circles into the day view (#397).
- Remove the redundant macro text row beneath the rings (#411).
- Calculations dialog split into focused entries for clearer settings (#414).

### Settings — diary day-start with minute granularity (#409)
- Configurable diary day-start (default 04:30) so the day boundary respects shift workers and people who eat past midnight.
- Used everywhere `today's` totals are computed.

### CSV export + JSON paste import
- Export bundle now ships CSV alongside the canonical JSON for intakes, activities, and tracked days (#398). Recipes ship JSON only (nested ingredient shape doesn't flatten cleanly).
- Paste a JSON blob from another instance to import meals (#395).
- Both documented in [`docs/export-format.md`](docs/export-format.md) (#393 documents the Supabase FDC self-hosting schema in the same sweep).

### Scanner — manual barcode entry (#425)
- Manual barcode entry shortcut when the camera struggles or the user is logging from a screenshot.
- Search bar accepts a pasted barcode directly.

### Meal detail — default to 1 serving when OFF provides it (#391)
- When OFF returns a `serving_quantity`, scanning a product defaults to 1 serving instead of forcing the user to convert from per-100g.

### Apple App Store guideline compliance (#416)
- Privacy manifest (`PrivacyInfo.xcprivacy`) declaring exactly what the app does and does not collect.
- **Delete all data** entry in Settings.
- Clearer onboarding consent text for crash reporting.
- Low-kcal warning when a goal calculation comes out unrealistically low.

### Sources & references (#380)
- In-app Sources & References screen one tap from the home calorie ring, the profile BMI card, the gender pickers, and the disclaimer dialog.
- Lists peer-reviewed sources for TDEE, BMI classification, macro distribution, MET activity calories, and non-binary calorie estimation.
- Closes Apple's 1.4.1 medical-citations finding.

### Project landing page (site/)
- Re-established the GitHub Pages landing site (#1439489e, #5e17e430).
- Site shows the signing-cert and APK fingerprints with an automation that refreshes them on each release (#0953933f).

## Localization

- **Slovak (`sk`)** added (#392) — full translations across the existing surface plus all the new strings below.
- Every new feature above ships its strings across the now-nine ARB files (en, de, cs, it, pl, **sk**, tr, uk, zh) plus matching getters in `lib/generated/l10n.dart` and per-locale `lib/generated/intl/messages_<locale>.dart`.

## Bug fixes

- iOS share sheet now actually presents, and the standard scanner recognises shared QRs (#440).
- Barcode scanner screens own and dispose their camera controller and release the camera when the app is backgrounded, fixing a leaked capture session and the dead torch / flip-camera buttons on the scan-product screen (#466).
- Onboarding shows possessive "What's" in height and weight prompts (#449).
- Show Activity Tracking toggle now hides the activity blocks consistently (#450).
- Sticky bottom buttons respect SafeArea (#394).
- Atwater-inconsistent nutrition entries surface a warning in edit-meal (#396).
- FDC import drops obviously-corrupt entries (#390).

### iOS
- Adaptive app-icon set (light / dark / tinted) shipped via Asset Catalog (#447).
- Privacy manifest and updated Info.plist for App Store compliance (#416).
- Podfile.lock refresh handling in CI (#628e215f), ios-podfile-lock-guard job ensures lockfile drift fails fast.

### Android
- Develop flavor wired into Gradle (#448) — `.develop` applicationId suffix means a develop-flavor APK installs alongside a Play Store install without overwriting it.
- Themed adaptive icon with monochrome layer (#447).
- Removed an unused permission (#ec90d9d3).

## CI / release pipeline

The big one: the **full release pipeline lands** from `integrate-cicd-deployment` (#439, with follow-ups #442). On every push to develop / main, the workflow now:

- Runs `linux-checks`, `ios-build`, `ios-podfile-lock-guard`, `android-build`, and (sharded — see #441) `ios-integration-tests` and `android-integration-tests` in parallel.
- On a main push (or workflow_dispatch), bundles signed iOS IPA via Fastlane → TestFlight, signed Android AAB/APK via Fastlane → Play Store internal track.
- Cuts a GitHub release tagged `vX.Y.Z` with both artifacts attached, on main only (#3076f18c, #e7e6b5d1).
- iOS deploy step moved to ubuntu-latest after the build job hands the IPA off (#e8322867) — saves macOS minutes.
- IPA package step uses FVM throughout instead of subosito's Flutter install (#852848d4) for parity with local dev.
- Build number is passed through fastlane (#ed20666e) so the CI run number is reflected in the uploaded artifact.

Integration test sharding (#441) discovers `integration_test/*.dart` at workflow time and fans them out across the matrix — one runner per file — gated on a clean iOS / Android build so we don't burn macOS minutes on a matrix when the app doesn't even compile.

Smaller CI fixes: Gemfile.lock for both ios and android (#31a12b10), setup-java + upload-artifact bump (#3f6239ce), cache@v5 (#bfb22058), checkout@v6 (#ff9fe20f), `cache-key-prefix` dropped from the shared `setup-flutter-cache` action (#b8142f92), per-job `timeout-minutes` dropped from the default workflow (#417).

## Testing + tooling

- ADB UI driver and per-branch smoke-test scripts (#386) plus hint-based form-field helpers with scroll + keyboard dismiss (#431) — `tools/adb/adb-driver.sh` is the entry point and is used by every per-branch verifier from here on.
- Semantics identifiers for ADB-driven UI testing (#384).
- Semantics(identifier:) convention documented in CLAUDE.md (#385).
- New unit tests cover the Quick add per-100g round trip, the meal-detail manual-add serving default, and several edge cases in the custom-meal Simple form.

## Documentation

- SHA256 signing fingerprint documented for APK side-loaders (#388, RELEASE.md).
- Supabase FDC self-hosting schema and refresh process documented (#393, `docs/supabase-fdc-self-hosting.md`).
- README, CONTRIBUTING, CLAUDE.md, and fastlane store copy all refreshed for this release (#457).

## Test plan

- [x] `just check_intl` passes (verifies generated `lib/generated/intl/*` matches ARBs)
- [x] `flutter analyze` clean
- [x] `just test` passes
- [ ] iOS smoke: TestFlight build launches; Quick add → log 500 kcal → home ring reflects it; water chip + fasting timer + weight chart visible; switch energy unit to kJ and confirm a diary entry renders correctly
- [x] Android smoke: Play Store internal-track AAB installs; same Quick add → water → fasting flow; themed app icon picks up the wallpaper accent
- [x] Migration smoke: upgrade an existing 1.3.x install — confirm intakes, custom meals, recipes, activities, and tracked days survive; new Water / Fasting / Weight log boxes open empty without crashing

## Post-merge follow-ups (@simonoppowa)

- The CI release pipeline on main now does the heavy lifting — but worth confirming:
  - `git tag v1.4.0 && git push origin v1.4.0` if the workflow's auto-tag didn't fire (it should — #3076f18c).
  - TestFlight build appears in App Store Connect → promote to public when ready.
  - Play Console → internal track build appears → promote through closed / open / production as your usual cadence.
- Store listing screenshots: the Recipes-tab and ingredient-picker screenshots from 1.3.0 still apply, but adding water, fasting, weight-history, and Material You panels would showcase the new headline features well.

### CODE REVIEW

**Comment by jordan-lee-code** [comment]
## End-to-end ADB testing — develop @ post-merge

Walked the v1.4.0+51 release candidate (`app-develop-debug.apk` from the
post-merge develop) on a Pixel 7 from a fresh install. Every tested
feature passes; nothing in the set below blocks the release.

### Onboarding intro now surfaces the sources link (#423, via merge PR #460)

Onboarding intro with sources link

The purple book icon + "Read our medical calculation sources" link sits
directly under the privacy consent checkboxes — fresh installs reach
it on the first onboarding page.

### Quick add — Title + Energy required, intake lands on the right meal (#451)

Greek yogurt 180 kcal quick-added to Breakfast

### Daily kcal banner on meal detail with two-tone projection bar (#452)

Day total 544 / 2410 with +364 kcal current selection annotation

### Water tracker — log dialog + chip updates (#418)

Water chip showing 250 / 1500 ml after a log

### Fasting timer — content-warning gate, then a running session, then a home chip (#432)

| Active timer view | Home chip |
| --- | --- |
| Fasting timer showing 15:59:21 remaining of a 16h target | Home chip reading "Fasting · 15h 58m left" |

### Weight history — entry saved, helper text honest about needing two days for a trend (#400)

Weight history with one entry, helper says log at least two days

### Recipes ingredient picker barcode scanner + manual entry fallback (#444 + #425)

| Scanner UI | Manual entry |
| --- | --- |
| Recipe ingredient barcode scanner with permission dialog | Enter barcode dialog with 8 to 14 digits hint |

### Settings — split into focused entries (#414)

Settings list showing Energy unit, Daily kcal adjustment, Macro split, Per-meal kcal share, Nutrient goals, Day starts at, etc

### Material You + 16-preset accent picker + custom hue (#426)

Use system colours toggle on, 16 colour circles in a grid, plus Custom colour entry

### kcal / kJ unit switch (#403)

Home ring rendering 9330 kJ left and 753 supplied (180 kcal converted)

### Per-meal kcal share — all five presets visible (#402)

Standard, Mediterranean, Two-meal, OMAD, Five-small presets plus per-meal sliders

### Micronutrient panel with IOM DRI bars (#404, #427)

All 10 nutrients with DRI reference values: fibre, sodium, sat fat, sugar, calcium, iron, potassium, vitamin D, vitamin B12, magnesium

### Diary sort by macro contribution (#405)

Sort menu with Time added, Calories, Protein, Carbs, Fat options

### Sources & References screen (#380)

Sources screen with intro, Energy needs IOM 2005, BMI WHO

### Day starts at — minute granularity (#409)

Day starts at dialog with Hours and Minutes sliders, current 0:00

### Custom meal Simple form, with photo and barcode on the same screen (#410 + #399 + #407)

Simple/Advanced toggle, Type the totals for one serving helper, photo placeholder, barcode field with scanner icon

### Custom activity — direct kcal entry + template (#408)

Custom activity screen with direct kcal entry and Save as template for next time checkbox

### Export — JSON / CSV toggle on the bundle (#398)

Export / Import App Data dialog with JSON tab selected, CSV tab visible

### Import Custom Food Data — CSV + paste-JSON (#395, #424)

Import Custom Food Data with CSV selected, JSON toggle visible, sample + import for meals and recipes

### Low-kcal warning — manually confirmed (#416)

Purple warning card on home: This daily target is on the low side. Most adults shouldn't eat fewer than 1200 kcal a day

Setting the daily kcal adjustment low enough to push the goal under
the 1200 kcal threshold reveals the App Store-compliance warning
exactly as designed.

### Things this harness can't cover

- The adaptive app icon (light / dark / tinted on iOS, themed on
  Android) — the icon resolves at the launcher level, not inside the
  app.
- Real barcode scanning — the camera was denied during the run; the
  manual-entry fallback exercises the same lookup path on the data side.
- The CI release pipeline itself — that will exercise on the merge of
  this PR.

### Recommendation

Ready for the release merge.

File: lib/core/domain/entity/config_entity.dart [hunk 1]
```diff
@@ -19,6 +39,74 @@ class ConfigEntity extends Equatable {
...
+  /// user has not picked one yet.
+  ///
+  /// The popular "8 × 8 oz = 2 L of plain water" target has no
+  /// scientific basis — Valtin (2002, AJP-Reg) traced it to a misreading
+  /// of a 1945 US Food and Nutrition Board note that explicitly said
+  /// "most of this quantity is contained in prepared foods". No major
+  /// guideline body recommends a single "2 L of plain water" figure
+  /// today.
+  ///
+  /// Picking 1500 ml lands inside the NHS Eatwell Guide's "6 to 8 cups
```
Comment (simonoppowa) [inline lib/core/domain/entity/config_entity.dart:82]: praise: really cool that you added scientific resources for the calculation as comments

File: lib/core/utils/bounds/ranges_const.dart [hunk 2]
```diff
@@ -9,6 +9,18 @@ class Ranges {
...
+  /// Soft floor below which a computed daily kcal goal is flagged as low
+  /// enough that the app surfaces a non-blocking warning encouraging the
+  /// user to consult a professional. Aligned with the commonly-cited
+  /// NIH / Mayo Clinic minimums for sustained healthy intake (≈1500 kcal
+  /// for adults with a testosterone-typical hormonal profile, ≈1200 kcal
+  /// otherwise). Non-binary users on the averaged midpoint, or who have
+  /// not yet chosen a calories profile, use the lower (female) floor
+  /// because the averaged TDEE is itself a midpoint and the lower bound
+  /// sits more naturally beside it.
+  static const double minRecommendedDailyKcalMale = 1500;
```
Comment (simonoppowa) [inline lib/core/utils/bounds/ranges_const.dart:23]: suggestion: i could not find any direct sources for these values from NIH or Mayo Clinic. However, we could add this link for reference from [Harvard Health](https://www.health.harvard.edu/healthy-aging-and-longevity/calorie-counting-made-easy)

File: lib/core/utils/calc/dri_reference.dart [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    _LifeStage.m71plus: DriReference(amount: 30, unit: 'g', basis: 'AI'),
+    _LifeStage.f19to30: DriReference(amount: 25, unit: 'g', basis: 'AI'),
+    _LifeStage.f31to50: DriReference(amount: 25, unit: 'g', basis: 'AI'),
+    _LifeStage.f51to70: DriReference(amount: 21, unit: 'g', basis: 'AI'),
+    _LifeStage.f71plus: DriReference(amount: 21, unit: 'g', basis: 'AI'),
+  },
+  NutrientPanelKeys.sodium: {
+    // IOM CDRR / UL: 2300 mg/day for all adults. The IOM does not publish
+    // a sodium RDA — the figure below is the chronic-disease-risk
+    // reduction intake, the same number the FDA Daily Value uses.
```
Comment (simonoppowa) [inline lib/core/utils/calc/dri_reference.dart]: suggestion: i checked the references for these values. They seem to be correct. But for the UL values i think it makes more sense to add a 'limit' label as they are not a recommended daily intake or use the AI values
I could also make sense to display AI and UL as an accepted range in the future

File: lib/core/utils/calc/dri_reference.dart [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    _LifeStage.m31to50: DriReference(amount: 2300, unit: 'mg', basis: 'UL'),
+    _LifeStage.m51to70: DriReference(amount: 2300, unit: 'mg', basis: 'UL'),
+    _LifeStage.m71plus: DriReference(amount: 2300, unit: 'mg', basis: 'UL'),
+    _LifeStage.f19to30: DriReference(amount: 2300, unit: 'mg', basis: 'UL'),
+    _LifeStage.f31to50: DriReference(amount: 2300, unit: 'mg', basis: 'UL'),
+    _LifeStage.f51to70: DriReference(amount: 2300, unit: 'mg', basis: 'UL'),
+    _LifeStage.f71plus: DriReference(amount: 2300, unit: 'mg', basis: 'UL'),
+  },
+  NutrientPanelKeys.calcium: {
+    // IOM RDA: 1000 mg adults 19 to 50 (and men 51 to 70),
```
Comment (simonoppowa) [inline lib/core/utils/calc/dri_reference.dart:136]: suggestion: the values are stated as AI in the reference as far as i see it

File: lib/core/utils/calc/dri_reference.dart [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    _LifeStage.m31to50: DriReference(amount: 8, unit: 'mg', basis: 'RDA'),
+    _LifeStage.m51to70: DriReference(amount: 8, unit: 'mg', basis: 'RDA'),
+    _LifeStage.m71plus: DriReference(amount: 8, unit: 'mg', basis: 'RDA'),
+    _LifeStage.f19to30: DriReference(amount: 18, unit: 'mg', basis: 'RDA'),
+    _LifeStage.f31to50: DriReference(amount: 18, unit: 'mg', basis: 'RDA'),
+    _LifeStage.f51to70: DriReference(amount: 8, unit: 'mg', basis: 'RDA'),
+    _LifeStage.f71plus: DriReference(amount: 8, unit: 'mg', basis: 'RDA'),
+  },
+  NutrientPanelKeys.potassium: {
+    // IOM AI (2019 update): men 3400 mg, women 2600 mg.
```
Comment (simonoppowa) [inline lib/core/utils/calc/dri_reference.dart:163]: suggestion: also link the 2019 update reference
https://www.nationalacademies.org/projects/HMD-FNB-17-01

File: lib/core/utils/calc/dri_reference.dart [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    _LifeStage.m51to70: DriReference(amount: 3400, unit: 'mg', basis: 'AI'),
+    _LifeStage.m71plus: DriReference(amount: 3400, unit: 'mg', basis: 'AI'),
+    _LifeStage.f19to30: DriReference(amount: 2600, unit: 'mg', basis: 'AI'),
+    _LifeStage.f31to50: DriReference(amount: 2600, unit: 'mg', basis: 'AI'),
+    _LifeStage.f51to70: DriReference(amount: 2600, unit: 'mg', basis: 'AI'),
+    _LifeStage.f71plus: DriReference(amount: 2600, unit: 'mg', basis: 'AI'),
+  },
+  NutrientPanelKeys.vitaminD: {
+    // IOM RDA: 15 µg (600 IU) adults to age 70, 20 µg (800 IU) 71+.
+    _LifeStage.m19to30: DriReference(amount: 15, unit: 'µg', basis: 'RDA'),
```
Comment (simonoppowa) [inline lib/core/utils/calc/dri_reference.dart:180]: suggestion: I can only find an AI in the table of the source (µg)

19 through 30 y 5
31 through 50 y 5
51 through 70 y 10 
70 y 15

File: lib/core/utils/notification_service.dart [hunk 7]
```diff
@@ -7,9 +7,14 @@ import 'package:timezone/timezone.dart' as tz;
 /// #312: Service for scheduling daily meal reminder notifications.
 class NotificationService {
   static const int _dailyReminderId = 0;
+  static const int _fastingCompleteId = 1;
   static const String _channelId = 'daily_reminder';
   static const String _channelName = 'Daily Reminders';
   static const String _channelDesc = 'Daily meal logging reminder';
+  static const String _fastingChannelId = 'fasting_complete';
+  static const String _fastingChannelName = 'Fasting timer';
```
Comment (simonoppowa) [inline lib/core/utils/notification_service.dart]: suggestion: All channel names and descriptions should be localised as there are shown to the user in the system settings

File: lib/features/activity_detail/activity_detail_screen.dart [hunk 8]
```diff
@@ -13,6 +17,7 @@ import 'package:opennutritracker/features/diary/presentation/bloc/calendar_day_b
 import 'package:opennutritracker/features/diary/presentation/bloc/diary_bloc.dart';
 import 'package:opennutritracker/features/home/presentation/bloc/home_bloc.dart';
 import 'package:opennutritracker/generated/l10n.dart';
+import 'package:provider/provider.dart';
 
 class ActivityDetailScreen extends StatefulWidget {
```
Comment (simonoppowa) [inline lib/features/activity_detail/activity_detail_screen.dart:22]: issue: the activity detail screen is still displaying '2024 Compendium ...' for custom activities

**Review by simonoppowa — COMMENTED** [review]
Amazing work 😁 So much cool stuff added.
I added some comments and will review the rest soon

File: lib/core/domain/entity/config_entity.dart (hunk 1 above)
Comment (jordan-lee-code) [reply lib/core/domain/entity/config_entity.dart:82]: Thank you, that means a lot. Keeping the citations next to the values felt like the right way to make them auditable later.

File: lib/core/utils/bounds/ranges_const.dart (hunk 2 above)
Comment (jordan-lee-code) [reply lib/core/utils/bounds/ranges_const.dart:23]: Good catch, I could not verify the NIH or Mayo figures either. In #470 I have swapped the comment to cite the Harvard Health page you linked.

File: lib/core/utils/calc/dri_reference.dart (hunk 3 above)
Comment (jordan-lee-code) [reply lib/core/utils/calc/dri_reference.dart]: Done in #470: the panel now shows a small "limit" label on the ceiling rows (sodium, saturated fat, sugar) so they no longer read as a target. I left the accepted-range idea for a future pass, as you suggested.

File: lib/core/utils/calc/dri_reference.dart (hunk 4 above)
Comment (jordan-lee-code) [reply lib/core/utils/calc/dri_reference.dart:136]: These come from the 2011 IOM calcium/vitamin D report, which set an EAR and RDA and replaced the 1997 AI, so the adult calcium figures are RDAs now. In #470 I added that note plus the source link inline. Happy to revisit if you are looking at a different table.

File: lib/core/utils/calc/dri_reference.dart (hunk 5 above)
Comment (jordan-lee-code) [reply lib/core/utils/calc/dri_reference.dart:163]: Added the 2019 update link to the potassium comment in #470, and to the sodium row too since the CDRR comes from the same report.

File: lib/core/utils/calc/dri_reference.dart (hunk 6 above)
Comment (jordan-lee-code) [reply lib/core/utils/calc/dri_reference.dart:180]: Those are the 1997 AI values; the 2011 IOM update established an RDA of 600 IU (15 µg) to age 70 and 800 IU (20 µg) for 71+, which is what NIH ODS publishes today. In #470 I kept the 2011 RDA so we are not surfacing older, lower numbers, and added the note and link at the line. Glad to talk it through if you would rather go the other way.

File: lib/core/utils/notification_service.dart (hunk 7 above)
Comment (jordan-lee-code) [reply lib/core/utils/notification_service.dart]: Done in #470: both channels now take their name and description from the user's locale across all nine languages, including the reboot-restore path in main(), so they read correctly in the system settings.

File: lib/features/activity_detail/activity_detail_screen.dart (hunk 8 above)
Comment (jordan-lee-code) [reply lib/features/activity_detail/activity_detail_screen.dart:22]: Fixed in #470: the Compendium attribution is now hidden for custom activities since they are user-entered. Thanks for spotting it.

**Review by simonoppowa — APPROVED** [review]
Looks good now 👍 Thanks for the great work on the project
